### PR TITLE
docs: first iteration of docs

### PR DESCRIPTION
**Pull request by tdejager**
## Description 

This adds the first documentation pages for the `build` preview feature. It adds another folder to the navigation, the page contains some explanations and a migration guide.

I've also found the dependency types reference lacking so made a seperate advanced page.

## Next up

* Changing the reference with new manifest keys.
* Some simple tutorials
* In-depth explanation for those interested in how the build system works.

### CODE REVIEW

File: docs/preview_features/build.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+We've been working to support these use-cases with the `build` feature in pixi. 
+The vision is to enable building of packages from source, for any language, on any platform.
+
+??? note "Preview feature"
+    This feature is currently in preview phase. To enable use the [preview feature](../reference/pixi_manifest.md#preview-features).
+    ```toml
+    [project]
+    # .. other configuration
+    preview = ["build"]
```
Comment (ruben-arts) [inline docs/preview_features/build.md:17]:
```suggestion
    preview = ["pixi-build"]
```
Maybe good to add these code examples to the manifests in `docs/source_files/pixi_tomls` So we automatically test them in ci.

File: docs/preview_features/build.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    ```
+
+## Limitations
+Currently, the `build` feature has a number of limitations:
+
+1. Recursive source dependencies are not supported.
```
Comment (ruben-arts) [inline docs/preview_features/build.md:23]:
```suggestion
1. Recursive source dependencies are not supported. ( source dependencies that have source dependencies )
```

File: docs/preview_features/build.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+2. Workspace dependencies cannot be inherited.
+3. Progress reporting should be improved.
+4. Only a limited number of languages and build systems are supported.
```
Comment (ruben-arts) [inline docs/preview_features/build.md:26]:
```suggestion
4. Limited set of [build-backends](https://github.com/prefix-dev/pixi-build-backends).
```

File: docs/preview_features/build.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+5. Missing parametrization of build-systems.
```
Comment (ruben-arts) [inline docs/preview_features/build.md:27]: Bas did implement arguments of some sort already right?

File: docs/preview_features/build.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+6. Using a feedstock directly as a source package is not supported.
+
+## High-level overview
+There are a couple of key concepts that make it easier to understand how the `build` feature works. The two most important
+additions are the concept of a *package* and a *build-system*.
+
+### Package
+The package defines the `build-system` and other fields in the future that are used to build your project.
+Currently, the dependencies, host-dependencies and other dependency fields are associated with the package.
+When you want to build a project you need to add this section.
```
Comment (ruben-arts) [inline docs/preview_features/build.md:37]: Which section? would be my question here. I think you meant `build-system` but i'm not sure.

**Review by ruben-arts — APPROVED** [review]
I did an initial review. I like the layout!

File: docs/preview_features/build.md (hunk 1 above)
Comment (tdejager) [reply docs/preview_features/build.md]: Does that work for partial tomls though?

File: docs/preview_features/build.md (hunk 4 above)
Comment (tdejager) [reply docs/preview_features/build.md:27]: Indeed, let me rephrase.

File: docs/preview_features/build.md (hunk 5 above)
Comment (tdejager) [reply docs/preview_features/build.md:37]: Let me just remove that line, I dont think it adds anything.

**Review by ruben-arts — APPROVED** [review]
Great!
